logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/receiver/asset/AssetMessageHandler.kt

```diff
@@ -28,21 +31,35 @@ internal class AssetMessageHandlerImpl(
         }
         val messageContent = message.content
         userConfigRepository.isFileSharingEnabled().onSuccess {
-            if (it.isFileSharingEnabled != null && it.isFileSharingEnabled) {
-                processNonRestrictedAssetMessage(message, messageContent)
-            } else {
-                val newMessage = message.copy(
-                    content = MessageContent.RestrictedAsset(
-                        mimeType = messageContent.value.mimeType,
-                        sizeInBytes = messageContent.value.sizeInBytes,
-                        name = messageContent.value.name ?: ""
-                    )
-                )
-                persistMessage(newMessage)
+            when (it.state) {
+                FileSharingState.Disabled -> {
+                    persistRestrictedAssetMessage(message, messageContent)
+                }
+
+                FileSharingState.EnabledAll -> processNonRestrictedAssetMessage(message, messageContent)
+
+                is FileSharingState.EnabledSome -> {
+                    if (validateAssetMimeTypeUseCase(messageContent.value.mimeType, it.state.allowedType)) {
+                        processNonRestrictedAssetMessage(message, messageContent)
+                    } else {
+                        persistRestrictedAssetMessage(message, messageContent)
+                    }
+                }
             }
```

Comment:
```suggestion
            val isThisAssetAllowed = when (it.state) {
                FileSharingState.Disabled -> false
                FileSharingState.EnabledAll -> true
                
                is FileSharingState.EnabledSome -> validateAssetMimeTypeUseCase(
                        messageContent.value.mimeType, 
                        it.state.allowedType)
                    )
            }
            
            if (isThisAssetAllowed) {
                processNonRestrictedAssetMessage(message, messageContent)
            } else {
                persistRestrictedAssetMessage(message, messageContent)
            }
```

A bit cleaner maybe?